src/python/gudhi/simplex_tree.pyx

```diff
@@ -184,6 +184,11 @@ cdef class SimplexTree:
         simplices are already present with a higher filtration value, their
         filtration value is lowered.
 
+        .. note::
+
+            Inserting a simplex with filtration value `math.nan` does not
+            modify the filtration value of any simplex already present.
```

Comment:
Perhaps complete this note by adding a warning in the vein of what is done for `assign_filtration()` saying (e.g.) "In this particular case, the result will not be a valid filtration anymore. Callers are responsible for fixing this (using `assign_filtration()` or `make_filtration_non_decreasing()` for instance) before calling any function that relies on the filtration property, like `persistence()`.

Reply:
I hadn't realized it, but having some NaN in the complex can have some weird effects. In particular, if I have vertex 0 with value NaN, inserting edge [0, 1] with value 42 does not change the value of vertex 0. I added some confusing text to the doc. We could change that behavior, but would need to check that it doesn't slow things down for normal operations.